test/types/connector.test-d.ts

```diff
@@ -19,3 +20,11 @@ expectAssignable<Client>(new Client('', {
     })
   }
 }))
+
```

Comment:
can we also have some more explicit tests for `TcpNetConnectOpts`  and `IpcNetConnectOpts` part?

Reply:
NetConnectOpts is `TcpNetConnectOpts | IpcNetConnectOpts`. I can make the tests for those separately if needed.

Reply:
it currently only has two fields, so I suspect if you remove some of the concatenated types, tests currently will not fail.
they should fail if any of concatenated types are removed

Reply:
Okay, so I ended up with:

```ts
expectAssignable<buildConnector.BuildOptions>({
  checkServerIdentity: () => undefined, // Test if ConnectionOptions is assignable
  localPort: 1234, // Check if TcpNetConnectOpts is assigable
});
```

The only problem here is that the only difference with `IpcNetConnectOpts` is that `port` is required. It does not have additional params as the other 2 do. Which would make adding `IpcNetConnectOpts` completely useless anyways probably, should I keep that one?

The ones I had indeed did not do anything.

Reply:
let's keep it, as conceptually we are supporting both, even if they overlap

Reply:
This one?
```ts
expectAssignable<buildConnector.BuildOptions>({
  checkServerIdentity: () => undefined, // Test if ConnectionOptions is assignable
  localPort: 1234, // Check if TcpNetConnectOpts is assigable
});
```
Or the one I had before?